src/Microsoft.Restier.Core/Query/DefaultQueryHandler.cs

```diff
@@ -58,7 +58,7 @@ internal static class DefaultQueryHandler
 
             // execute query
             QueryResult result;
-            var executor = context.GetApiService<IQueryExecutor>();
+            var executor = context.GetApiContextService<IQueryExecutor>();
```

Comment:
Most of Restier services have singleton lifetime, so only very few objects will get created and disposed for a new scope.